src/aggregation_rules/aligned_time_rule.rs

```diff
@@ -96,4 +97,43 @@ mod tests {
         assert_eq!(c.open(), 13768.5);
         assert_eq!(c.close(), 13722.0);
     }
+
+    #[test]
+    fn aligned_time_rule_trigger_on_0() {
+        let trades: [Trade; 5] = [
+            Trade {
+                timestamp: 1712656800000,
+                price: 100.0,
+                size: 10.0,
+            },
+            Trade {
+                timestamp: 1712656815000,
+                price: 101.0,
+                size: -10.0,
+            },
+            Trade {
+                timestamp: 1712656860000,
+                price: 101.0,
+                size: -10.0,
+            },
+            Trade {
+                timestamp: 1712656860001,
+                price: 102.0,
+                size: -10.0,
+            },
+            Trade {
+                timestamp: 1712656935000,
+                price: 105.0,
+                size: -10.0,
+            },
+        ];
+
+        let mut aggregator = GenericAggregator::<OhlcCandle, AlignedTimeRule, Trade>::new(
+            AlignedTimeRule::new(M1, TimestampResolution::Millisecond),
+        );
+        let candles = aggregate_all_trades(&trades, &mut aggregator);
+        assert_eq!(candles.len(), 2);
+        assert_eq!(candles[0].open(), 100.00);
+        assert_eq!(candles[1].open(), 101.00);
```

Comment:
```suggestion
        assert_eq!(candles[0].open(), 100.0);
        assert_eq!(candles[0].close(), 100.5);
        assert_eq!(candles[1].open(), 101.0);
        assert_eq!(candles[1].close(), 101.0);
```
An assertion for the close price would be good as well.
I think those values are correct, but unfortunate that both open and close prices are 101, so maybe it makes sense to add another trade to be included in the second candle